main.c

```diff
@@ -74,10 +74,11 @@ static struct tcmur_handler *find_handler_by_subtype(gchar *subtype)
 
 int tcmur_register_handler(struct tcmur_handler *handler)
 {
-	if (handler->handle_cmd &&
-	    (handler->read || handler->write || handler->flush)) {
-		tcmu_err("Skip bad handler: %s\n", handler->name);
-		return -1;
+	if (handler->read || handler->write || handler->flush) {
+		if (handler->handle_cmd && !handler->cmdset) {
```

Comment:
A check for the inverse is also required (to be safe), else it will segfault during `->handler_cmd()`